internal/controller/pkg/revision/dependency.go

```diff
@@ -92,8 +93,18 @@ func (m *PackageDependencyManager) Resolve(ctx context.Context, pkg runtime.Obje
 
 	// Get the lock.
 	lock := &v1alpha1.Lock{}
-	if err := m.client.Get(ctx, types.NamespacedName{Name: lockName}, lock); err != nil {
-		return found, installed, invalid, errors.Wrap(err, errGetLock)
+	err = m.client.Get(ctx, types.NamespacedName{Name: lockName}, lock)
```

Comment:
Can we use `Apply` here?

Reply:
@muvaf we perform operations based on the content of the Lock in the case it exists here so I don’t think so, but maybe I am not understanding your suggestion.